Automattic/wp-calypso
# i18n: Link to translated content where possible

While we do have translated Privacy Policies to a few languages but we have those links hardcoded to the English version only. Thus users (rightfully) believe we only provide them in English.

This PR introduces a new function that allows to pass through the English URLs and rewrites them if applicable. You can think of this as a `translate()` for URLs.

Effectively this aims to relieve the developer from having to think about the right URL for a language. `localizeUrl` is meant to do the right thing for you.

Examples:
```
> getLocaleSlug()
'en'
> localizeUrl( 'https://automattic.com/privacy/' );
'https://automattic.com/privacy/'
> localizeUrl( 'https://wordpress.com/tos/' );
'https://wordpress.com/tos/'
> localizeUrl( 'https://en.support.wordpress.com/' );
'https://en.support.wordpress.com/'
> localizeUrl( 'https://en.forums.wordpress.com/' );
'https://en.forums.wordpress.com/'
```

```
> getLocaleSlug()
'ru'
> localizeUrl( 'https://automattic.com/privacy/' );
'https://automattic.com/privacy/' # Privacy Policy not translated to Russian
> localizeUrl( 'https://wordpress.com/tos/' );
'https://ru.wordpress.com/tos/'
> localizeUrl( 'https://en.support.wordpress.com/' );
'https://ru.support.wordpress.com/'
> localizeUrl( 'https://en.forums.wordpress.com/' );
'https://ru.forums.wordpress.com/'
```

```
> getLocaleSlug()
'de'
> localizeUrl( 'https://automattic.com/privacy/' );
'https://automattic.com/de/privacy/'
> localizeUrl( 'https://wordpress.com/tos/' );
'https://de.wordpress.com/tos/'
> localizeUrl( 'https://en.support.wordpress.com/' );
'https://de.support.wordpress.com/'
> localizeUrl( 'https://en.forums.wordpress.com/' );
'https://de.forums.wordpress.com/'
```
```
> getLocaleSlug()
'th'
> localizeUrl( 'https://automattic.com/privacy/' );
'https://automattic.com/privacy/'
> localizeUrl( 'https://wordpress.com/tos/' );
'https://wordpress.com/tos/'
> localizeUrl( 'https://en.support.wordpress.com/' );
'https://en.support.wordpress.com/'
> localizeUrl( 'https://en.forums.wordpress.com/' );
'https://th.forums.wordpress.com/' # We do have Thai forums
```

Task list:

- [x] create `localizeUrl()`
- [x] pass links to https://automattic.com/privacy through `localizeUrl()`
- [x] pass links to https://automattic.com/cookies through `localizeUrl()`
- [x] pass links to https://wordpress.com/tos/ through `localizeUrl()`
- [x] replace all usages of `getSupportSiteLocale()` with `localizeUrl()`
- [x] replace all usages of `getForumUrl()` with `localizeUrl()`
- [x] replace all usages of `addLocaleToWpcomUrl()` with `localizeUrl()`

#### Reviewing Instructions
The meat of this is in [lib/i18n-utils/utils.js](https://github.com/Automattic/wp-calypso/pull/27857/files#diff-138fcb256fa5ea3ba188a240f6f007ee). Most of other files referenced functions replaced by `localizeUrl()`.

#### Testing instructions

Go to https://calypso.localhost:3000/me/privacy and check the links. Depending on the locale, the language version should be linked. Currently translated are German, French, and Spanish.

`npm run test-client lib/i18n-utils`
